modules/OFConnectionManager2/module/src/ofconnectionmanager.c

```diff
@@ -0,0 +1,1774 @@
+ *
+ * Listening connections may not receive all async messages.
+ *
+ * Currently, derived listening controllers are not tracked, and
+ * there is no cleanup of derived controllers/connections
+ * if the main listening controller is removed.
+ *
+ * When the listening controller is removed, the main listening
+ * connection is stopped by calling controller_stop_cxn,
+ * which in turn calls ind_cxn_stop.  The connection's state advances
+ * from initialized (since the connection state machine is not
+ * used by the listener) to closing, and then immediately to closed.
+ * This will close the underlying socket and unregister from 
+ * socket manager.
+ */
+
+/**
+ * Handle a listening socket ready event
+ *
+ * @param socket_id The socket that is ready
+ * @param cookie Pointer to the connection instance
+ * @param read_ready Was read-ready signaled by select
+ * @param write_ready Was write-ready signaled by select
+ * @param error_seen Was an error signaled by select
+ *
+ * This usually means a new socket connection is ready to be
+ * accepted.  Spin up a new connection instance for it.
+ */
+static void
+listen_socket_ready(int socket_id, void *cookie, int read_ready,
+                    int write_ready, int error_seen)
+{
+    connection_t *listen_cxn;
+    socklen_t addrlen;
+    struct sockaddr_in cxn_addr;
+    int new_sd;
+    connection_t *cxn;
+    indigo_controller_id_t controller_id;
+    controller_t *controller;
+
+    listen_cxn = (connection_t *)cookie;
+
+    AIM_LOG_TRACE("Listen soc %d ready, cxn %p. rd %d. wr %d. er %d",
+                  socket_id, cookie, read_ready, write_ready, error_seen);
+
+    if (!CXN_ID_VALID(listen_cxn->cxn_id)) {
+        AIM_LOG_INTERNAL("Listen socket ready with bad cxn %p", listen_cxn);
+        return;
+    }
+
+    if (!CXN_ACTIVE(listen_cxn)) {
+        AIM_LOG_INTERNAL("Listen socket ready on non active cxn %p",
+                         listen_cxn);
+        return;
+    }
+
+    AIM_ASSERT(listen_cxn->controller != NULL);
+    AIM_ASSERT(listen_cxn->sd == socket_id);
+
+    if (error_seen) {
+        int socket_error = 0;
+        socklen_t len = sizeof(socket_error);
+        getsockopt(listen_cxn->sd, SOL_SOCKET, SO_ERROR, &socket_error, &len);
+        AIM_LOG_ERROR("listen %s: %s", 
+                      listen_cxn->desc, strerror(socket_error));
+        return;
+    }
+
+    /* Ready for an accept */
+    if (!read_ready) {
+        AIM_LOG_ERROR("listen %s: read not ready",
+                      listen_cxn->desc);
+        return;
+    }
+
+    /* Accept the new client */
+    addrlen = sizeof(cxn_addr);
+    new_sd = accept(listen_cxn->sd, (struct sockaddr *)&cxn_addr,
+                    &addrlen);
+    if (new_sd == -1) {
+        AIM_LOG_ERROR("Accept on listen %s: %s",
+                      listen_cxn->desc, strerror(errno));
+        return;
+    }
+    AIM_LOG_VERBOSE("Accepted cxn on port %d", ntohs(cxn_addr.sin_port));
```

Comment:
this looks historical.  i've added some code that will print the peer address for ipv4, and a fixme for ipv6 (since we don't have an aim datatype for ipv6 yet.